totalRP3/totalRP3.lua

```diff
@@ -104,6 +104,29 @@ local function loadingSequence()
 
 	TRP3_API.events.fireEvent(TRP3_API.events.NAVIGATION_RESIZED, TRP3_MainFramePageContainer:GetWidth(), TRP3_MainFramePageContainer:GetHeight());
 
+	-- Hack to implement the ROLEPLAY_STATUS_CHANGED event; we'll monitor
+	-- REGISTER_DATA_UPDATED for player profile changes and trigger the event
+	-- if the roleplay status changes.
+	local currentRoleplayStatus; -- Initialized to nil explicitly.
+
+	TRP3_API.events.registerCallback(TRP3_API.events.REGISTER_DATA_UPDATED, function(unitID)
+		-- Disregard updates for non-player changes.
+		if unitID ~= TRP3_API.globals.player_id then
+			return;
+		end
+
+		local currentUser = AddOn_TotalRP3.Player.GetCurrentUser();
+		local isInCharacter = currentUser:IsInCharacter();
+
+		if currentRoleplayStatus ~= isInCharacter then
+			currentRoleplayStatus = isInCharacter;
+			TRP3_API.events.triggerEvent(TRP3_API.events.ROLEPLAY_STATUS_CHANGED);
```

Comment:
Could you pass the new roleplay status as a parameter to the event since you already have it here, so future listeners don't have to also go look for it?

Reply:
I thought about it, but I had reservations. The event signals that the status has changed, but the only value I can meaningfully pass through at the moment is a boolean (true for IC, false for OOC). 

Passing a boolean doesn't feel particularly obvious from the standpoint of using that event if you think it out loud; "okay, the roleplay status has changed and my new state is... `true`?".

If a payload were to be added I'd prefer it to send a token like "in_character" or "out_of_character" that's a bit more descriptive of the actual state.

Reply:
You could add something to our enums and pass that value, so it can be used on the other ends for comparing. Numbers are probably lighter than strings for that kind of values.

Reply:
Or even use booleans for the values, to make it "compatible" with all existing functions